### PR TITLE
[gh-actions] replace local setup-lxd action with canonical/setup-lxd

### PR DESCRIPTION
**Pull request by barrettj12**
The setup-lxd action added in #14912 has now been moved into its own repo, so we can delete the local copy.

## QA steps

Ensure all checks pass on this PR.

### CODE REVIEW

**Review by hpidcock — APPROVED** [review]

**Comment by barrettj12** [comment]
Have some updates to make to the canonical/setup-lxd action, then I'll pull in an updated version and merge.

File: .github/workflows/smoke.yml [hunk 1]
```diff
@@ -36,7 +35,7 @@ jobs:
       uses: actions/checkout@v3
 
     - name: Setup LXD
-      uses: ./.github/setup-lxd
+      uses: canonical/setup-lxd@90d76101915da56a42a562ba766b1a77019242fd
```
Comment (juanmanuel-tirado) [inline .github/workflows/smoke.yml:38]: I know this has already been merged. Specially, for this reason, this should target a released version, not a SHA.

Comment (barrettj12) [reply .github/workflows/smoke.yml:38]: doesn't have any releases yet, maybe will add one when it proves stable enough.

**Pull request merged**